pkg/datastore/pipedstore.go

```diff
@@ -224,6 +224,7 @@ func (s *pipedStore) UpdateDesiredVersion(ctx context.Context, id, version strin
 
 func (s *pipedStore) UpdateMetadata(ctx context.Context, id, version, config string, pps []*model.Piped_PlatformProvider, repos []*model.ApplicationGitRepository, se *model.Piped_SecretEncryption, startedAt int64) error {
 	return s.update(ctx, id, func(piped *model.Piped) error {
+		piped.CloudProviders = nil
```

Comment:
I have one point I would like to confirm.
Is this safe for the user using only `cloudProviders` settings in piped?

Reply:
The piped execution only refers to its local configuration and itself knows about the CloudProvider (suppose it has only CloudProviders and the piped runs the old version since the new one will merge all CloudProviders -> PlatformProviders), so I think it's okay since they know which "cloudProvider" to use on its side.

Otherwise, though we remove CloudProviders information from the piped model here, it only be used to show on the application adding form selection, so I think it's okay 🤔

Reply:
I got your point! Thanks.